source/base/subscriptor.cc

```diff
@@ -175,7 +158,6 @@ Subscriptor::subscribe(const char *id) const
 #ifdef DEBUG
   if (object_info == 0)
     object_info = &typeid(*this);
-  Threads::Mutex::ScopedLock lock (subscription_lock);
   ++counter;
```

Comment:
I have no idea (but, to be fair, I don't have a lot of experience using threads; perhaps this means something to an expert?).

I'll post again when I have added a comment in the source and updated the description in the header.